app/Rules/ValidTimezone.php

```diff
@@ -18,438 +19,20 @@ public function validate(string $attribute, mixed $value, Closure $fail): void
-            'Asia/Pyongyang' => 'Pyongyang (UTC+09:00)',
-            'Asia/Seoul' => 'Seoul (UTC+09:00)',
-            'Asia/Tokyo' => 'Tokyo (UTC+09:00)',
-            'Australia/Adelaide' => 'Adelaide (UTC+09:30)',
-            'Australia/Broken_Hill' => 'Broken Hill (UTC+09:30)',
-            'Australia/Darwin' => 'Darwin (UTC+09:30)',
-            'Australia/Brisbane' => 'Brisbane (UTC+10:00)',
-            'Pacific/Chuuk' => 'Chuuk (UTC+10:00)',
-            'Australia/Currie' => 'Currie (UTC+10:00)',
-            'Antarctica/DumontDUrville' => 'DumontDUrville (UTC+10:00)',
-            'Pacific/Guam' => 'Guam (UTC+10:00)',
-            'Australia/Hobart' => 'Hobart (UTC+10:00)',
-            'Asia/Khandyga' => 'Khandyga (UTC+10:00)',
-            'Australia/Lindeman' => 'Lindeman (UTC+10:00)',
-            'Australia/Melbourne' => 'Melbourne (UTC+10:00)',
-            'Pacific/Port_Moresby' => 'Port Moresby (UTC+10:00)',
-            'Pacific/Saipan' => 'Saipan (UTC+10:00)',
-            'Australia/Sydney' => 'Sydney (UTC+10:00)',
-            'Asia/Yakutsk' => 'Yakutsk (UTC+10:00)',
-            'Australia/Lord_Howe' => 'Lord Howe (UTC+10:30)',
-            'Pacific/Efate' => 'Efate (UTC+11:00)',
-            'Pacific/Guadalcanal' => 'Guadalcanal (UTC+11:00)',
-            'Pacific/Kosrae' => 'Kosrae (UTC+11:00)',
-            'Antarctica/Macquarie' => 'Macquarie (UTC+11:00)',
-            'Pacific/Noumea' => 'Noumea (UTC+11:00)',
-            'Pacific/Pohnpei' => 'Pohnpei (UTC+11:00)',
-            'Asia/Sakhalin' => 'Sakhalin (UTC+11:00)',
-            'Asia/Ust-Nera' => 'Ust-Nera (UTC+11:00)',
-            'Asia/Vladivostok' => 'Vladivostok (UTC+11:00)',
-            'Pacific/Norfolk' => 'Norfolk (UTC+11:30)',
-            'Asia/Anadyr' => 'Anadyr (UTC+12:00)',
-            'Pacific/Auckland' => 'Auckland (UTC+12:00)',
-            'Pacific/Fiji' => 'Fiji (UTC+12:00)',
-            'Pacific/Funafuti' => 'Funafuti (UTC+12:00)',
-            'Asia/Kamchatka' => 'Kamchatka (UTC+12:00)',
-            'Pacific/Kwajalein' => 'Kwajalein (UTC+12:00)',
-            'Asia/Magadan' => 'Magadan (UTC+12:00)',
-            'Pacific/Majuro' => 'Majuro (UTC+12:00)',
-            'Antarctica/McMurdo' => 'McMurdo (UTC+12:00)',
-            'Pacific/Nauru' => 'Nauru (UTC+12:00)',
-            'Antarctica/South_Pole' => 'South Pole (UTC+12:00)',
-            'Pacific/Tarawa' => 'Tarawa (UTC+12:00)',
-            'Pacific/Wake' => 'Wake (UTC+12:00)',
-            'Pacific/Wallis' => 'Wallis (UTC+12:00)',
-            'Pacific/Chatham' => 'Chatham (UTC+12:45)',
-            'Pacific/Apia' => 'Apia (UTC+13:00)',
-            'Pacific/Enderbury' => 'Enderbury (UTC+13:00)',
-            'Pacific/Fakaofo' => 'Fakaofo (UTC+13:00)',
-            'Pacific/Tongatapu' => 'Tongatapu (UTC+13:00)',
-        ];
+        $timezones = DateTimeZone::listIdentifiers(DateTimeZone::ALL_WITH_BC);
+
+        return array_flip($timezones);
```

Comment:
why?

Reply:
Because if we use `in_array()` to search for values it is slow if we compare it to `isset` or `empty()`.

Reply:
Don't you think it can throw an undefined array key exception?

Reply:
nevermind